lib/react_client.dart

```diff
@@ -734,8 +746,54 @@ class ReactJsComponentFactoryProxy extends ReactComponentFactoryProxy {
   }
 }
 
+class ReactDartFunctionComponentFactoryProxy extends ReactComponentFactoryProxy
+    with JsBackedMapsComponentFactoryBuilder {
+  /// The ReactJS function used as the type for all [ReactElement]s built by
+  /// this factory.
+  final JsFunctionComponent reactFunction;
+
+  /// The name of this function.
+  final String displayName;
+
+  /// The JS component factory used by this factory to build [ReactElement]s.
+  final ReactJsComponentFactory reactComponentFactory;
+
+  ReactDartFunctionComponentFactoryProxy(this.reactFunction, [this.displayName])
+      : this.reactComponentFactory = React.createFactory(reactFunction);
+
+  get type => reactFunction;
+}
+
+/// Creates a function component from the given [dartFunctionComponent] that can be used with React.
+JsFunctionComponent _wrapFunctionComponent(DartFunctionComponent dartFunctionComponent, {String displayName}) {
+  assert(() {
+    displayName ??= getProperty(dartFunctionComponent, 'name');
+    return true;
+  }());
+  jsFunctionComponent(JsMap jsProps, [JsMap _legacyContext]) =>
+      (dartFunctionComponent(jsProps != null ? JsBackedMap.backedBy(jsProps) : JsBackedMap()) ?? jsNull);
```

Comment:
We should have a line comment linking to the undefined/null issue as an explanation for `?? jsNull`